server/src/main/java/org/apache/druid/segment/realtime/SegmentGenerationMetrics.java

```diff
@@ -55,6 +55,65 @@ public class SegmentGenerationMetrics
 
   private final AtomicLong maxSegmentHandoffTime = new AtomicLong(NO_EMIT_SEGMENT_HANDOFF_TIME);
 
+  public static class MessageGapStats
+  {
+    long minMessageGap = Long.MAX_VALUE;
+    long maxMessageGap = Long.MIN_VALUE;
+    long numMessageGap = 0;
+    double totalMessageGap = 0;
+
+    public synchronized double avgMessageGap()
+    {
+      return totalMessageGap / numMessageGap;
+    }
+
+    public synchronized long getMinMessageGap()
+    {
+      return minMessageGap;
+    }
+
+    public synchronized long getMaxMessageGap()
+    {
+      return maxMessageGap;
+    }
+
+    public synchronized long getNumMessageGap()
+    {
+      return numMessageGap;
+    }
+
+    public synchronized void add(final long messageGap)
+    {
+      totalMessageGap += messageGap;
+      ++numMessageGap;
+      if (minMessageGap > messageGap) {
+        minMessageGap = messageGap;
+      }
+      if (maxMessageGap < messageGap) {
+        maxMessageGap = messageGap;
+      }
+    }
+
+    public MessageGapStats getAndReset()
```

Comment:
This method should not be needed, use an `AtomicReference` in `SegmentGenerationMetrics` instead.